### PR TITLE
Update of the classifier model

### PR DESCRIPTION
**Pull request by jsschreck**
Charlie's suggestions from previous PR:

- Can remove the classifier argument from the init as it is not used
- The callbacks and annealing_coefficient is a bit wonky. Currently with a coefficient of 34, for example, you need both annealing_coefficient=34 and callbacks=[ReportEpoch(34)] which is awkward. Perhaps you add the callback behind the scenes but still allow other callbacks to be appended.
- Currently, to use the evidential model, you need to set the loss="dirichlet" and output_activation="linear". It would be nice if we could maybe label the loss "evidential" and do the output layer in the background.
- The class_weights argument has no effect (even setting them all to zero produces same output). I know that line is commented out here, but I have tested this. I do not understand why it doesn't work. Also, in some of the other model classes, you use the weights with the argument loss_weights= in the model compile stage - I don't believe this is the same as class_weights in the fit stage. I will make a separate comment.
- The loss_weights argument here is not the same as class_weights in the .fit() method which I think is your intention.
- In the other model classes, I think the calc_prob_uncertainties() is a method within the class, and here it is a function outside the class. It would be better if they were consistent.

### CODE REVIEW

File: mlguess/keras/models.py [hunk 1]
```diff
@@ -1241,27 +1268,24 @@ def predict_ensemble(self, x, weight_locations, batch_size=None):
...
-def calc_prob_uncertainty(y_pred, num_classes=4):
-    evidence = tf.nn.relu(y_pred)
-    alpha = evidence + 1
-    S = tf.keras.backend.sum(alpha, axis=1, keepdims=True)
-    u = num_classes / S
-    prob = alpha / S
-    epistemic = prob * (1 - prob) / (S + 1)
-    aleatoric = prob - prob**2 - epistemic
-    return prob, u, aleatoric, epistemic
+    def predict_uncertainty(self, x, num_classes=4):
```
Comment (djgagne) [inline mlguess/keras/models.py:1279]: You should be able to get num_classes from y_pred.shape[1] rather than having the user specify it and potentially get an incorrect result if they forget to change the default.

**Review by djgagne — CHANGES_REQUESTED** [review]

**Review by djgagne — APPROVED** [review]
It looks like you addressed the main issues here. Merging now.
